recipes/small_gicp/all/conanfile.py

```diff
@@ -0,0 +1,124 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import copy, get, rm, rmdir, download
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class IridescenceConan(ConanFile):
+    name = "small_gicp"
+    description = "Efficient and parallelized algorithms for point cloud registration"
+    license = "MIT AND BSD" # BSD is from nanoflann and Sophus
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/koide3/small_gicp"
+    topics = ("point-cloud", "icp", "registration", "scan-matching", "pcl")
+
+    package_type = "library" # TODO: could add a header_only option as well
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openmp": [True, False],
+        "with_tbb": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openmp": True,
+        "with_tbb": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "7",
+            "apple-clang": "10",
+            "Visual Studio": "15",
+            "msvc": "191",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("eigen/3.4.0", transitive_headers=True)
+        if self.options.with_openmp:
+            # '#pragma omp' is used in public headers
+            # Note: native MSVC OpenMP is not compatible
+            self.requires("llvm-openmp/18.1.8", transitive_headers=True, transitive_libs=True)
```

Comment:
Can we `validate()` this configuration out to be able to move this forward while work on mp progresses?